test/suites/integration/conversion_webhook_test.go

```diff
@@ -0,0 +1,239 @@
+									"memory.available":   {Duration: time.Minute * 2},
+									"nodefs.available":   {Duration: time.Minute * 2},
+									"nodefs.inodesFree":  {Duration: time.Minute * 2},
+									"imagefs.available":  {Duration: time.Minute * 2},
+									"imagefs.inodesFree": {Duration: time.Minute * 2},
+									"pid.available":      {Duration: time.Minute * 2},
+								},
+								EvictionMaxPodGracePeriod:   lo.ToPtr(int32(120)),
+								ImageGCHighThresholdPercent: lo.ToPtr(int32(50)),
+								ImageGCLowThresholdPercent:  lo.ToPtr(int32(10)),
+								CPUCFSQuota:                 lo.ToPtr(false),
+							},
+						},
+					},
+				},
+			}
+
+			// Use a deepcopy to make sure the nodePool object is not populated with the returned object from the APIServer
+			env.ExpectCreated(storedv1beta1NodePool.DeepCopy())
+			v1beta1NodePool := env.ExpectExists(storedv1beta1NodePool.DeepCopy()).(*karpv1beta1.NodePool)
+			Expect(v1beta1NodePool.Generation).To(BeNumerically("==", 1))
+
+			// Second apply of the same NodePool does not increase the generation
+			env.ExpectUpdated(storedv1beta1NodePool.DeepCopy())
+			v1beta1NodePool = env.ExpectExists(storedv1beta1NodePool.DeepCopy()).(*karpv1beta1.NodePool)
+			Expect(v1beta1NodePool.Generation).To(BeNumerically("==", 1))
+			Expect(v1beta1NodePool.Spec.Template.Spec.NodeClassRef.APIVersion).To(Equal(storedv1beta1NodePool.Spec.Template.Spec.NodeClassRef.APIVersion))
+			Expect(v1beta1NodePool.Spec.Disruption.ConsolidateAfter.Duration.String()).To(Equal(storedv1beta1NodePool.Spec.Disruption.ConsolidateAfter.Duration.String()))
+			Expect(v1beta1NodePool.Spec.Disruption.ExpireAfter.Duration.String()).To(Equal(storedv1beta1NodePool.Spec.Disruption.ExpireAfter.Duration.String()))
+			// Kubelet Validation
+			Expect(v1beta1NodePool.Spec.Template.Spec.Kubelet.MaxPods).To(Equal(storedv1beta1NodePool.Spec.Template.Spec.Kubelet.MaxPods))
+			Expect(v1beta1NodePool.Spec.Template.Spec.Kubelet.PodsPerCore).To(Equal(storedv1beta1NodePool.Spec.Template.Spec.Kubelet.PodsPerCore))
+			Expect(v1beta1NodePool.Spec.Template.Spec.Kubelet.SystemReserved).To(Equal(storedv1beta1NodePool.Spec.Template.Spec.Kubelet.SystemReserved))
+			Expect(v1beta1NodePool.Spec.Template.Spec.Kubelet.KubeReserved).To(Equal(storedv1beta1NodePool.Spec.Template.Spec.Kubelet.KubeReserved))
+			Expect(v1beta1NodePool.Spec.Template.Spec.Kubelet.EvictionHard).To(Equal(storedv1beta1NodePool.Spec.Template.Spec.Kubelet.EvictionHard))
+			Expect(v1beta1NodePool.Spec.Template.Spec.Kubelet.EvictionSoft).To(Equal(storedv1beta1NodePool.Spec.Template.Spec.Kubelet.EvictionSoft))
+			Expect(v1beta1NodePool.Spec.Template.Spec.Kubelet.EvictionSoftGracePeriod).To(Equal(storedv1beta1NodePool.Spec.Template.Spec.Kubelet.EvictionSoftGracePeriod))
+			Expect(v1beta1NodePool.Spec.Template.Spec.Kubelet.EvictionMaxPodGracePeriod).To(Equal(storedv1beta1NodePool.Spec.Template.Spec.Kubelet.EvictionMaxPodGracePeriod))
```

Comment:
considering these are the only ones with non-primitive types, how can we ensure we're testing the deserialized values as opposed to the String() values? How are we testing equality here?

Reply:
We are only testing to nillable duration, and don't need to ensure this for metav1.Duration

Reply:
Ah sure makes sense